doc/decisions/0_drafts/key_value.md

```diff
@@ -0,0 +1,123 @@
+# `Key` value API
+
+## Problem
+
+A `Key` has a value, that much is clear.
+From the ["`Key` value types" decision](../3_decided/key_value_types.md) we know that all values are opaque byte-sequences of a known length.
+
+But how should the API for accessing and modifying the value of a `Key` look?
```

Comment:
What is a problem if not a question without an  answer?

Anyway, the point of this decision is to decide the future API for key values. I did not mention any problems with the current API, because:
1. As soon as the decision is implemented that makes the problem harder to understand IMO. I'd have to give a detailed description of the current API, because people could no longer (easily) check what the current API is doing.
2. Because of other decisions (specifically the one in this PR, but also COW to an extent) the current API is no longer ideal. It was not designed with these other decisions in mind.